### DISCUSSION TITLE
Building a lab with Pykka?

### DISCUSSION THREAD

**Discussion by gaelgrissonnanche** [jodal/pykka#214] [Q&A]
Hi,
I am building a lab and would like to know if this is a good long-term strategy for using Pykka for this development.
The number of active contributors is one, which seems dangerous. I am always worried about poorly maintained packages or on this slippery slope. What would you say is the vitality of this package?
However, your work is amazing and Pykka would fulfill all my lab's present needs, it is the future I am worried about.

**Comment by jodal** [comment]
Hi Gael!

I don't know exactly what you mean by building a lab, so I can't address that use case in particular.

In general I can say that it is safe to rely on Pykka for its current feature set, as I've already been maintaining the project with this feature set for 15 years. Pykka is extensively used in the Mopidy music server ecosystem, so it is not going anywhere.

That said, I do not recommend relying on Pykka if you think that you'll need to get requests for new features accepted and merged within reasonable time.

Tl;dr: Pykka is "done", but it is maintained.

**Reply by gaelgrissonnanche** [reply]
Hi!

Thank you very much for your answer. 
I am sorry for not being more precise, I mean that I am building a research lab in which we will communicate with a large number of scientific instruments. I come from Labview where the Actor paradigm is very well developed to discuss with instruments, and I am surprised that your package is not more widely used in research for people using Python.
My question was indeed to know if Pykka would stay around and be maintained for a while, a question that is hard to answer I understand, but I am a bit hesitant to rely on it fully for instrument communication and see this jewel disappear one day...which will likely not be the case of standard libraries in Python. 
I really find your package very convenient and much lighter to use than implementing myself Threads, Events and Queues or use Asyncio in the program.

**Reply by jodal** [reply]
It's great to hear that you get value from Pykka and like the abstraction!

I wouldn't be too worried of Pykka becoming unmaintained or disappear, as it only relies on the Python standard library and comes with a test suite with 100% line coverage, so it should be fairly easy to keep working indefinitely for any Python developer.

**Reply by gaelgrissonnanche** [reply]
Perfect, thank you!
